### PR TITLE
Update Israel holidays

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Update Israel holidays:
- holiday categories support, move some holidays to OPTIONAL and SCHOOL
- add Sigd, Tisha B'Av, Jerusalem Day, Fast of Esther (optional)
- rewrite tests

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've added references to all holidays information sources used in this PR
- [x] The code style looks good: `make pre-commit` command generates no changes
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Comment by KJhellico** [comment]
Next step is l10n. :)

File: holidays/countries/israel.py [hunk 1]
```diff
@@ -9,96 +9,159 @@
...
 
-        # Passover
-        passover_dt = _HebrewLunisolar.hebrew_holiday_date(year, "PASSOVER")
-        self._add_with_eve("Passover I", passover_dt)
-        for offset in range(1, 5):
-            self._add_holiday("Passover - Chol HaMoed", passover_dt + td(days=offset))
-        self._add_with_eve("Passover VII", passover_dt + td(days=+6))
+        # Yom Kippur
+        yom_kippur_dt = _HebrewLunisolar.hebrew_holiday_date(self._year, "YOM_KIPPUR")
+        self._add_holiday("Yom Kippur", yom_kippur_dt)
```
Comment (arkid15r) [inline holidays/countries/israel.py:52]: Could you inline values used only once?
```suggestion
        self._add_holiday("Yom Kippur", self._get_holiday("YOM_KIPPUR"))
```

**Review by arkid15r — COMMENTED** [review]
Good stuff 👍

**Review by arkid15r — APPROVED** [review]
🇮🇱 LGTM

**Pull request merged**